dist/primitive_check.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+# This is a temporary script to detect code changes to WiredTiger primitives.
+
+import subprocess, re
+primitives = [
+    "WT_BARRIER",
+    "WT_WRITE_BARRIER",
+    "WT_PUBLISH",
+    "WT_READ_BARRIER",
+    "WT_ORDERED_READ",
+    "WT_FULL_BARRIER",
+    "__wt_atomic_.*",
+    "FLD_SET_ATOMIC",
+    "FLD_CLR_ATOMIC",
+    "F_SET_ATOMIC",
+    "F_CLR_ATOMIC",
+    "WT_PAGE_ALLOC_AND_SWAP",
+    "WT_REF_UNLOCK",
+    "WT_REF_LOCK",
+    "WT_REF_CAS_STATE",
+    "REF_SET_STATE",
+    "WT_INTL_INDEX_SET",
+    "volatile",
+    "WT_STAT_INCRV_ATOMIC",
+    "WT_STAT_DECRV_ATOMIC",
+    "WT_DHANDLE_ACQUIRE",
+    "WT_DHANDLE_RELEASE",
+    "FLD_SET_ATOMIC",
+    "FLD_CLR_ATOMIC"
+]
+
+diff = subprocess.run(['git', 'diff'], capture_output=True, text=True).stdout
+found = False
+found_primitives = []
+start_regex = "(\+|-).*"
```

Comment:
Suggested regex: `^(\+|-).*` to guarantee this pattern is at the start of this line